packages/@aws-cdk/aws-cloudfront/lib/distribution.ts

```diff
@@ -3,6 +3,7 @@ import { Construct, IResource, Lazy, Resource, Stack, Token, Duration } from '@a
 import { CfnDistribution } from './cloudfront.generated';
 import { Origin } from './origin';
 import { CacheBehavior } from './private/cache-behavior';
+import { LambdaFunctionAssociation } from './web_distribution';
```

Comment:
Two options here:
1. I've been trying to move the shared enums into this file, so that later when we deprecate the CloudFrontWebDistribution, it will be clear that everything in `web_distribution` is deprecated, and everything in `distribution` is the new hotness.
2. It's a little weird for the property to be `edgeLambdas` and the interface to be `LambdaFunctionAssociation`. Maybe we should just create a new `EdgeLambda` interface (with the `[Edge]EventType` enum)?

Reply:
Ok. I've moved `LambdaEdgeEventType` from `web_distribution.ts` to `distribution.ts`, and created a new `EdgeLambda` interface with 2 properties, `functionVersion` and `eventType`.

Let me know if this works better!